src/main/java/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.java

```diff
@@ -178,11 +217,14 @@ public Querier encrypt(int numThreads) throws InterruptedException, PIRException
       {
         // Hash collision
         selectorQueryVecMapping.clear();
-        hashKey = queryInfo.getHashKey() + ++keyCounter;
+        hashKey = hashKeyBase + getRandByteString(10);
         logger.debug("index = " + index + "selector = " + selector + " hash collision = " + hash + " new key = " + hashKey);
         index = -1;
       }
     }
+    
+    // Save off the final hashKey that we ended up using
+    queryInfo.setHashKey(hashKey);
```

Comment:
The ```QueryInfo``` is an API type, so the end user is free to reuse it (and encrypt a query using it) as often as required.

Let me try a different angle ;-)  Why do we ask for a user to specify a hashKey anyway?  Why not simply generate a totally random hashKey rather than a salted key?